src/decoder.js

```diff
@@ -1,59 +1,186 @@
-import consensysDecode from 'abi-decoder';
-import { isAddress } from './web3/utils';
-
-const decodedAddressList = [];
-
-/**
- * Return added ABI addresses
- * @return Array
- */
-export const getDecodedAddresses = () => decodedAddressList;
-
-/**
- * Add ABI for Decoding system
- *
- * @param Object abi
- * @param string address
- * @return Object
- */
-export const addABI = (abi, address) => {
-  if (!isAddress(address)) throw new Error('Input is not an address');
-  if (!abi) throw new Error('can\'t add undefined/null ABI');
-  decodedAddressList.push(address);
-  consensysDecode.addABI(abi);
-};
-
-/**
- * Decode Transaction input data
- *
- * @param Object inputData
- * @return Object
- */
-export const decodeInputData = (inputData) => {
-  if (!consensysDecode.getABIs().length) throw new Error('No ABIs added to system');
-  if (inputData === '0x') return ''; // no need for decoding if inputData is empty
-  if (!inputData) throw new Error('decodedData is undefined/invalid');
-  const decodedData = consensysDecode.decodeMethod(inputData);
-  if (!decodedData || decodedData === undefined) {
-    return { name: 'UNDECODED', params: [{ name: 'rawData', value: inputData, type: 'data' }] };
+import SolidityCoder from 'web3/lib/solidity/coder';
+import web3 from './web3/web3Provider';
+
+export default class Decoder {
+  /**
+   * Initialize class local variables
+   * @param Array abiArray
+   */
+  constructor(abiArray) {
+    this.abiArray = abiArray;
+    this.methodIDs = {};
+    if (Array.isArray(this.abiArray)) {
+      // Iterate new abi to generate method id's
+      this.abiArray.map((abi) => {
```

Comment:
use the map function when you are going to save the results into variable, in this case just use foreach or normal for loop.